circle.yml

```diff
@@ -4,96 +4,138 @@
+    - run:
+        working_directory: shared
+        command:  yarn run rn-gobuild-android
+    - run: ls -la $GOPATH/src/github.com/keybase/client/shared/react-native/android/keybaselib/keybaselib.aar
+
     # Build a debug version of the app
     # Bundle JS separately because doing it in the gradle build is buggy
-    - mkdir -p react-native/android/app/build/intermediates/assets/releaseUnsigned
-    - mkdir -p react-native/android/app/build/intermediates/res/merged/releaseUnsigned
-    - react-native bundle --verbose --platform android --dev false --entry-file index.android.js --bundle-output react-native/android/app/build/intermediates/assets/releaseUnsigned/index.android.bundle --assets-dest react-native/android/app/build/intermediates/res/merged/releaseUnsigned:
+    - run: mkdir -p shared/react-native/android/app/build/intermediates/{assets,res/merged}/releaseUnsigned
+    - run:
+        working_directory: shared
         environment:
-          PATH: $HOME/.yarn/bin:$PATH
+          REACT_NATIVE_MAX_WORKERS: 1
+        command: react-native bundle --verbose --platform android --dev false --entry-file index.android.js --bundle-output react-native/android/app/build/intermediates/assets/releaseUnsigned/index.android.bundle --assets-dest react-native/android/app/build/intermediates/res/merged/releaseUnsigned
+
     # Build app
-    - ./gradlew assembleReleaseUnsigned -x bundleReleaseUnsignedJsAndAssets:
-        pwd:
-          react-native/android
-test:
-  override:
-    - mkdir -p $CIRCLE_TEST_REPORTS/flow-coverage
-    - "true"
-    # TODO(mm) uncomment these device tests (see todo above)
-    # wait for it to have booted
-    # - circle-android wait-for-boot
-    # run tests  against the emulator.
-    # - (cd android && ./gradlew connectedAndroidTest)
-    # copy the build outputs to artifacts
-    # - cp -r android/app/build/outputs $CIRCLE_ARTIFACTS
-    # copy the test results to the test results directory.
-    # - cp -r android/app/build/outputs/androidTest-results/* $CIRCLE_TEST_REPORTS
-  post:
-      # Only run coverage if there were changes to the shared folder
-    - if [ $(git rev-parse HEAD) == $(git log -1 --format=format:%H --full-diff .) ]; then yarn flow-coverage-report -t json -o $CIRCLE_TEST_REPORTS/flow-coverage && bash <(curl -s https://codecov.io/bash) -f $CIRCLE_TEST_REPORTS/flow-coverage/flow-coverage.json; fi
+    - run:
+        working_directory: shared/react-native/android
+        environment:
+          TERM: dumb
+          # Limit memory usage of gradle
+          _JAVA_OPTIONS: -Xms512m -Xmx2048m
+          GRADLE_OPTS: -Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"
+        command: ./gradlew assembleReleaseUnsigned -x bundleReleaseUnsignedJsAndAssets
+
+    # Save dependency cache
+    - save_cache:
+        key: v1-dep-{{ .Branch }}-{{ epoch }}
+        paths:
+        - ~/.gradle
+        - ~/downloads
+        - ~/.cache/yarn
+        - /tmp/go-android/bin
```

Comment:
Not sure what we have in these two dirs

Reply:
You can do an ssh run and see. Caching the go tmp dir speeds up android to build